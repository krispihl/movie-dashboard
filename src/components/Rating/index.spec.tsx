import { Rating } from "./index";
import { render } from '@testing-library/react';

describe('src/components/Rating', () => {
    it('should render 0 Rating correctly', async () => {
        const { container } = render(<Rating rating={0} />);

        const icons = Array.from(document.querySelectorAll('.material-icons'));

        icons.map(icon => {
            expect(icon).toHaveTextContent('star_outline');
        })

        expect(container).toMatchSnapshot();
    });

    it('should render 5 Rating correctly', async () => {
        const { container } = render(<Rating rating={5} />);

        const icons = document.querySelectorAll('.material-icons');

        expect(icons[0]).toHaveTextContent('star');
        expect(icons[1]).toHaveTextContent('star');
        expect(icons[2]).toHaveTextContent('star_half');
        expect(icons[3]).toHaveTextContent('star_outline');
        expect(icons[4]).toHaveTextContent('star_outline');

        expect(container).toMatchSnapshot();
    });

    it('should render 10 Rating correctly', async () => {
        const { container } = render(<Rating rating={10} />);

        const icons = Array.from(document.querySelectorAll('.material-icons'));

        icons.map(icon => {
            expect(icon).toHaveTextContent('star');
        })

        expect(container).toMatchSnapshot();
    })
});
