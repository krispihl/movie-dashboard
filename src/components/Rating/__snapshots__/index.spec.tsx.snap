// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Rating should render 0 Rating correctly 1`] = `
<div>
  <ul>
    <span
      class="material-icons icon"
    >
      star_outline
    </span>
    <span
      class="material-icons icon"
    >
      star_outline
    </span>
    <span
      class="material-icons icon"
    >
      star_outline
    </span>
    <span
      class="material-icons icon"
    >
      star_outline
    </span>
    <span
      class="material-icons icon"
    >
      star_outline
    </span>
  </ul>
</div>
`;

exports[`src/components/Rating should render 5 Rating correctly 1`] = `
<div>
  <ul>
    <span
      class="material-icons icon"
    >
      star
    </span>
    <span
      class="material-icons icon"
    >
      star
    </span>
    <span
      class="material-icons icon"
    >
      star_half
    </span>
    <span
      class="material-icons icon"
    >
      star_outline
    </span>
    <span
      class="material-icons icon"
    >
      star_outline
    </span>
  </ul>
</div>
`;

exports[`src/components/Rating should render 10 Rating correctly 1`] = `
<div>
  <ul>
    <span
      class="material-icons icon"
    >
      star
    </span>
    <span
      class="material-icons icon"
    >
      star
    </span>
    <span
      class="material-icons icon"
    >
      star
    </span>
    <span
      class="material-icons icon"
    >
      star
    </span>
    <span
      class="material-icons icon"
    >
      star
    </span>
  </ul>
</div>
`;
